lms/static/scripts/frontend_apps/components/LMSFilePicker.tsx

```diff
@@ -184,18 +184,53 @@ export default function LMSFilePicker({
     }
   };
 
+  // Does the actual file fetching for the active directory, either by loading
+  // it from its children, or by going to the API to get them
+  const loadFilesToDisplay = useCallback(
+    (isReload = false): Promise<File[]> => {
+      const getNextAPICallInfo = () =>
+        folderPath[folderPath.length - 1]?.contents || listFilesApi;
+      const loadFilesFromApi = (): Promise<File[]> =>
+        apiCall({
+          authToken,
+          path: getNextAPICallInfo().path,
+        });
+
+      if (!isReload) {
+        // If the last element in current folderPath contains children, we use
+        // them without calling the API. Otherwise, we fetch files from the API
+        const children = folderPath[folderPath.length - 1]?.children;
+        return children ? Promise.resolve(children) : loadFilesFromApi();
+      }
+
+      return loadFilesFromApi().then(loadedFiles => {
```

Comment:
Oh! I should have called this out in the last round, but since we did the review verbally I missed it. Our convention is not to mix `Promise` chaining with `async/await` code where possible (we have a general preference for `async` over `Promises` when feasible).

Could you make `loadFilesToDisplay`'s callback and `loadFilesFromApi` both `async` and `await loadedFiles = loadFilesFromApi();`? It also makes this chunk of code easier to scan and reason about.